dsgrid/utils/spark_partition.py

```diff
@@ -10,21 +10,37 @@ class SparkPartition:
     def __init__(self):
         return
 
-    def get_data_size(self, df, bytes_per_cell=64):
+    def get_data_size(self, df, bytes_per_cell=8):
         """ approximate dataset size """
         n_rows = df.count()
         n_cols = len(df.columns)
         data_MB = n_rows * n_cols * bytes_per_cell / 1e6  # MB
         return n_rows, n_cols, data_MB
 
     @timed_info
-    def get_optimal_number_of_files(self, df, MB_per_file=128):
-        """ calculate *optimal* number of files """
+    def get_optimal_number_of_files(self, df, MB_per_cmp_file=128, cmp_ratio=0.18):
+        """calculate *optimal* number of files
+        Parameters
+        ----------
+        df : DataFrame
+        MB_per_cmp_file : float
+            Size of compressed file in MB
```

Comment:
Change to, "Desired size of compressed file on disk in MB"?

Also, how would one determine cmp_ratio? Would it be possible to make a temp file that serializes a small portion of the data in order to calculate the cmp_ratio?

@daniel-thom

Reply:
If the data has floats without repeating values, this is likely pretty good. I can imagine getting a better ratio with 8-byte floats over 4-byte floats, so that could be checked. If the data has large runs of repeated values then the ratio will be huge. IIRC, at least one version of the comstock data exhibits this. If we use this function on the one-table format then this applies to the dimension columns.

All that being said, I think the effort to optimize this probably isn't worth it.

Reply:
We could do that but it's not super necessary because testing compression on a small snippet of the data is still a guessing game and compression ratio is near the ballpark of the default in most cases. The function is meant to be used interactively and iteratively until the actual saved output is of an acceptable size.